controlplane/kubeadm/controllers/scale.go

```diff
@@ -90,8 +90,8 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 ) (ctrl.Result, error) {
 	logger := controlPlane.Logger()
 
-	if err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); err != nil {
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
+	if result, err := r.reconcileHealth(ctx, cluster, kcp, controlPlane); !result.IsZero() || err != nil {
+		return result, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
```

Comment:
Do you still want this returning a RequeueAfterError?

Reply:
Same as above